### PR TITLE
Better telemetry docs

### PR DESCRIPTION
**Pull request by ryanuber**
### What

Adds details for telemetry data, including logs, tracing, and metrics. This change attempts to give good coverage of current metrics, without including any details which are subject to change, such as the tags and labels applied to each metric. This is intended to provide a basic inventory of metrics and high-level descriptions of what each measurement means.

### Why

Multiple users have asked for this, and this follows the same general pattern for documenting metrics as taken by other products such as Nomad or Consul.

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [ ] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [ ] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/agents/telemetry.mdx [hunk 1]
```diff
@@ -4,40 +4,46 @@ page_title: Telemetry - Terraform Cloud Agents - Terraform Cloud and Terraform E
...
+[OpenTelemetry](https://opentelemetry.io/) protocol. The agent is a simple
+producer of telemetry data, and relies on an OpenTelemetry Collector to perform
+aggregation, post-processing, and exporting of telemetry data to upstream
+technology and/or vendors. OpenTelemetry is an extensible telemetry protocol and
+open source platform, which enables the Terraform Cloud Agent to remain
+vendor-agnostic while producing high-value telemetry data for a wide array
+of platforms.
 
-Optionally, you can pass the `-otlp-cert-file` or `TFC_AGENT_OTLP_CERT_FILE`. The agent will use a certificate at the path supplied to encrypt the client connection to the OpenTelemetry collector. When omitted, client connections are not secured.
+![telemetry architecture diagram](../../../img/teletry-arch.svg)
```
Comment (clairevnext) [inline website/docs/cloud-docs/agents/telemetry.mdx:21]: This link is broken. I think there is one too many `../`

**Comment by clairevnext** [comment]
This is an amazing amount of data to get documented! 🎉 Thank you for all the time this must have taken.

File: website/docs/cloud-docs/agents/telemetry.mdx (hunk 1 above)
Comment (ryanuber) [reply website/docs/cloud-docs/agents/telemetry.mdx]: Woops, it would also help if I spelled `telemetry` correctly 🤦‍♂️ Thanks!

File: website/docs/cloud-docs/agents/telemetry.mdx [hunk 2]
```diff
@@ -4,40 +4,104 @@ page_title: Telemetry - Terraform Cloud Agents - Terraform Cloud and Terraform E
...
+of the OpenTelemetry client libraries built in. As these libraries and
+implementations change, it is possible that certain versions of the agent will
+become incompatible with newer versions of the OpenTelemetry collector. To
+track version compatibility, the table below will be updated whenever these
+version constraints become known.
 
-An example of the metric names that the agent will emit is `tfc-agent.terraform.plan-json.generate.bytes`. Breaking down those sections:
+| Terraform Cloud Agent version | Compatible OpenTelemetry Collector version |
+| ----------------------------- | ------------------------------------------ |
+| All versions                  | <= 0.48.0                                  |
```
Comment (zisom-hc) [inline website/docs/cloud-docs/agents/telemetry.mdx:59]: although 0.48.0 is _technically_ compatible with the current libraries used, i remember that not all data was making it to Datadog, or GCP when attempting to utilize this version, and it wasn't until i made it down to `0.42.0`, the version we use internally, that i was able to see all telemetry making it successfully into DD at the least. We may want to change this to 0.42.0, as it's the best known version to work with how we have things instrumented.

File: website/docs/cloud-docs/agents/telemetry.mdx [hunk 3]
```diff
@@ -4,40 +4,104 @@ page_title: Telemetry - Terraform Cloud Agents - Terraform Cloud and Terraform E
 
 # Telemetry
 
-Terraform Cloud Agents emit telemetry data using the [OpenTelemetry](https://opentelemetry.io/) protocol.
+Terraform Cloud Agents may be configured to emit helpful telemetry data,
+including both trace spans and metrics.
```
Comment (zisom-hc) [inline website/docs/cloud-docs/agents/telemetry.mdx:8]: the instrumentation technically has the ability to emit logging as well, and is enabled/configured to be possible within the TFC agent currently, right? I can't recall if i've verified this prior.

**Review by zisom-hc — CHANGES_REQUESTED** [review]
In total, all of this new verbosity is great; thank you Ryan for the time put into consolidating all this information. Here are my edits/suggestions:

File: website/docs/cloud-docs/agents/telemetry.mdx [hunk 4]
```diff
@@ -4,40 +4,104 @@ page_title: Telemetry - Terraform Cloud Agents - Terraform Cloud and Terraform E
...
 
-In addition, agent metrics will follow some conventions around unit types:
+By default, the `contrib` distribution accepts telemetry from a wide variety of
+sources, and dumps information about the metrics and traces collected to stdout.
+To start the collector with default settings, run the following command:
 
-* All timing metrics (other than runtime metrics) will be measured in milliseconds.
-* All data size metrics will be measured in bytes.
+```
+docker run -p 127.0.0.1:4317:4317/tcp otel/opentelemetry-collector-contrib:latest
```
Comment (zisom-hc) [inline website/docs/cloud-docs/agents/telemetry.mdx:68]: i notice `latest` being configured in the command. I suppose what this should be will get complicated as we have newer version of the TFC agent that support newer instrumentation libraries, but maybe we should:

* make this `0.42.0` for now
* Forego a configuration of that in the example command (Which would still default to latest)
* Configure it as X.X.X, and change the instruction to say to replace the X's with the appropriate collector version, using the chart within the compatibility section; something along those lines

File: website/docs/cloud-docs/agents/telemetry.mdx [hunk 5]
```diff
@@ -4,40 +4,104 @@ page_title: Telemetry - Terraform Cloud Agents - Terraform Cloud and Terraform E
...
+
+To tweak the configuration, save your custom YAML configuration to a file on
+disk called `collector.yaml`, and run the collector container with the
+following command:
+
+```
+docker run \
+  --mount type=bind,source=${PWD}/collector.yml,target=/etc/otelcol-contrib/config.yaml \
+  -p 127.0.0.1:4317:4317/tcp \
+  otel/opentelemetry-collector-contrib:latest
```
Comment (zisom-hc) [inline website/docs/cloud-docs/agents/telemetry.mdx:89]: i notice `latest` being configured in the command. I suppose what this should be will get complicated as we have newer version of the TFC agent that support newer instrumentation libraries, but maybe we should:

* make this `0.42.0` for now
* Forego a configuration of that in the example command (Which would still default to latest)
* Configure it as X.X.X, and change the instruction to say to replace the X's with the appropriate collector version, using the chart within the compatibility section; something along those lines

**Review by zisom-hc — CHANGES_REQUESTED** [review]
Comments about the example commands

File: website/docs/cloud-docs/agents/tracing.mdx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+The agent emits tracing spans that can be consumed by various distributed
+tracing tools. Information about supported tools can be found on the
+[OpenTelemetry Registry](https://opentelemetry.io/registry/). A span is a
+single unit of work performed by the agent.
+
+Spans conform to the following rules:
+
+* Each span has a unique name in plain English, explaining what it represents.
+  Due to this descriptive nature of spans, individual trace spans are not
```
[the remark this answers is not in the document]
Comment (ryanuber) [reply website/docs/cloud-docs/agents/tracing.mdx:15]: This is the type of documentation that would go stale very easily, which is why I opted to not include it. I think the maintenance burden ends up outweighing the utility here. It's similar to not documenting every log line - I believe it is overly verbose, and the user should be able to get context on any given trace frame from looking at the frame itself.  Metrics are a bit different, as users are likely to build monitors around them and depend on them to some extent, and there is not a good cross-platform way of ensuring they provide accurate and understandable context without additional documentation.

File: website/docs/cloud-docs/agents/logging.mdx [hunk 7]
```diff
@@ -6,11 +6,70 @@ description: >-
 
 # Logging
 
-You may want to monitor Terraform Cloud Agent output.
+Terraform Cloud Agents log helpful messages which enable operators to understand
+agent behavior, including communication with Terraform Cloud APIs, specific
+commands run and actions taken, and runtime management.
```
[the remark this answers is not in the document]
Comment (ryanuber) [reply website/docs/cloud-docs/agents/logging.mdx:11]: It's probably better to remove the extra "and". Thanks for the suggestion!

File: website/docs/cloud-docs/agents/telemetry.mdx (hunk 4 above)
Comment (ryanuber) [reply website/docs/cloud-docs/agents/telemetry.mdx:68]: Oh great call-out, thank you. I like the third option best as it allows us to refer to the compatibility chart, and doesn't force us to update the example commands as things change.

File: website/docs/cloud-docs/agents/telemetry.mdx (hunk 3 above)
Comment (ryanuber) [reply website/docs/cloud-docs/agents/telemetry.mdx:8]: We do not use OpenTelemetry's logging facilities at all, primarily because even at the protocol level they are still [marked as experimental](https://opentelemetry.io/docs/reference/specification/logs/).

File: website/docs/cloud-docs/agents/telemetry.mdx (hunk 2 above)
Comment (ryanuber) [reply website/docs/cloud-docs/agents/telemetry.mdx:59]: Let's go with 0.42.0 then 👍

File: website/docs/cloud-docs/agents/metrics.mdx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+## Terraform component metrics
+
+The following metrics are emitted by the `terraform` component, which is
+responsible for handling Terraform operations like plans and applies. All
+metrics in this section are prefixed by `tfc-agent.terraform.`.
+
+| Meric name                                     | Type  | Description                                                                         |
+| ---------------------------------------------- | ----- | ----------------------------------------------------------------------------------- |
+| `handle-signal.milliseconds`                   | Timer | Time spent handling an incoming signal from Terraform Cloud.                        |
```
[the remark this answers is not in the document]
Comment (ryanuber) [reply website/docs/cloud-docs/agents/metrics.mdx:86]: I think we can wordsmith this a little more. Let's see...

File: website/docs/cloud-docs/agents/telemetry.mdx (hunk 4 above)
Comment (ryanuber) [reply website/docs/cloud-docs/agents/telemetry.mdx:68]: On second thought though, this makes it more difficult to just copy and go, especially because this is a multi-line command (per guidelines for code in docs). I think for now, I'll make this 0.42.0, so that it works. If we can get `latest` working soon, we can update back to `latest`.

**Comment by ryanuber** [comment]
Thanks so much for the thorough read, @zisom-hc ! I applied your suggestions.

File: website/docs/cloud-docs/agents/telemetry.mdx (hunk 4 above)
Comment (zisom-hc) [reply website/docs/cloud-docs/agents/telemetry.mdx]: that sounds good, and 🤞 for the potential utilization of latest soon!

File: website/docs/cloud-docs/agents/tracing.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (zisom-hc) [reply website/docs/cloud-docs/agents/tracing.mdx:15]: That's fair. If any customers feel different we can get a count of that through a Feature Request at the least.

**Review by zisom-hc — APPROVED** [review]
new edits LGTM, 🚢

**Review by natalie-todd — APPROVED** [review]
I added a couple of tiny spelling nits. 

This context/documentation is extremely helpful for the `tfc-agent` telemetry project. Thank you!

**Review by judithpatudith — APPROVED** [review]
Thank you!
